### PR TITLE
Render primitive schema data & enums in the worker inspector

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description

This PR adds a subset of the component data rendering to the Worker Inspector. It adds support for all schema primitives (except for `Entity`) and schema enums. Later PRs will add

- user defined types
- schema collections (lists, maps, and options)
- the `Entity` schema type

Note in https://github.com/spatialos/gdk-for-unity/pull/1387/commits/f0816334405202b840b365a4ecf92eec682f7a5d I turned quite a few things public. Discussed a strategy with Paul and will raise a ticket to go back and refactor these to make using them easier in the modular codegen world. 

image

#### Tests

Eyeball test. Will conduct more exhaustive tests once all the data types are in

### CODE REVIEW

File: workers/unity/Packages/io.improbable.gdk.debug/.codegen/Source/Generators/DebugExtensions/ComponentVisualElementGenerator.cs [hunk 1]
```diff
@@ -22,28 +29,184 @@ public static CodeWriter Generate(UnityComponentDetails details)
                     {
                         type.Line($"public override ComponentType ComponentType {{ get; }} = ComponentType.ReadOnly<{details.Name}.Component>();");
 
+                        type.TextList(TextList.New(details.FieldDetails.Select(ToFieldDeclaration)));
```
Comment (paulbalaji) [inline workers/unity/Packages/io.improbable.gdk.debug/.codegen/Source/Generators/DebugExtensions/ComponentVisualElementGenerator.cs:32]: Might be worth adding a variant of TypeBlock.TextList that takes an IEnumerable and does TextList.New for you under the hood?

File: workers/unity/Packages/io.improbable.gdk.debug/.codegen/Source/Generators/DebugExtensions/ComponentVisualElementGenerator.cs [hunk 2]
```diff
@@ -22,28 +29,184 @@ public static CodeWriter Generate(UnityComponentDetails details)
...
         private static void GenerateConstructor(TypeBlock typeBlock, UnityComponentDetails details)
         {
             typeBlock.Method($"public {details.Name}Renderer() : base()", mb =>
             {
                 mb.Line($"ComponentFoldout.text = \"{details.Name}\";");
                 mb.Line($"AuthoritativeToggle.SetEnabled(false);");
+
+                foreach (var field in details.FieldDetails)
+                {
+                    mb.TextList(TextList.New(ToFieldInitialisation(field)));
```
Comment (paulbalaji) [inline workers/unity/Packages/io.improbable.gdk.debug/.codegen/Source/Generators/DebugExtensions/ComponentVisualElementGenerator.cs:70]: same comment but for methods too

File: workers/unity/Packages/io.improbable.gdk.debug/.codegen/Source/Generators/DebugExtensions/ComponentVisualElementGenerator.cs (hunk 1 above)
Comment (jamiebrynes7) [reply workers/unity/Packages/io.improbable.gdk.debug/.codegen/Source/Generators/DebugExtensions/ComponentVisualElementGenerator.cs:32]: Yeah that would be a bit more ergonomic. There's an `ICollection` version I think, but not `IEnumerable`

File: workers/unity/Packages/io.improbable.gdk.debug/.codegen/Source/Generators/DebugExtensions/ComponentVisualElementGenerator.cs [hunk 3]
```diff
@@ -22,28 +29,184 @@ public static CodeWriter Generate(UnityComponentDetails details)
...
+                    var uiType = GetUiFieldType(singularFieldType.ContainedType);
+
+                    if (uiType == "")
+                    {
+                        // TODO: Eliminate this case.
+                        yield break;
+                    }
+
+                    var humanReadableName = Formatting.SnakeCaseToHumanReadable(fieldDetails.Name);
+                    yield return $"{fieldDetails.CamelCaseName}Field = new {uiType}(\"{humanReadableName}\");";
```
Comment (zeroZshadow) [inline workers/unity/Packages/io.improbable.gdk.debug/.codegen/Source/Generators/DebugExtensions/ComponentVisualElementGenerator.cs:90]: Extract the string into a variable: `{fieldDetails.CamelCaseName}Field`

Comment (jamiebrynes7) [reply workers/unity/Packages/io.improbable.gdk.debug/.codegen/Source/Generators/DebugExtensions/ComponentVisualElementGenerator.cs:90]: I've refactored this a little bit in the user-types branch, will apply this suggestion there 👍

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/FieldTypes/ContainedType.cs [hunk 4]
```diff
@@ -6,27 +6,27 @@ namespace Improbable.Gdk.CodeGeneration.Model.Details
     public class ContainedType
     {
         public readonly string FqnType;
-        private readonly ValueType category;
-        private PrimitiveType? primitiveType;
+        public readonly ValueType Category;
+        public PrimitiveType? PrimitiveType;
```
[the remark this answers is not in the document]
Comment (jamiebrynes7) [reply workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/FieldTypes/ContainedType.cs]: Ye can be

**Pull request merged**